src/modules/lua.c

```diff
@@ -73,11 +73,17 @@ mtev_lua_lmc_alloc(mtev_dso_generic_t *self, mtev_lua_resume_t resume) {
 void
 mtev_lua_lmc_free(lua_module_closure_t *lmc) {
   if(lmc) {
+    if(lmc->lua_state) lua_close(lmc->lua_state);
```

Comment:
The lmc is thread local.. that state will only ever be seen/manipulated by a single thread. so I think we're safe to do this locklessly.